tablo/models.py

```diff
@@ -432,25 +451,32 @@ def _validate_where_clause(self, where):
         parsed = self._parse_where_clause(where)
 
         if parsed is None:
-            return True
+            return
         elif parsed[1]:
-            return False  # More than one statement means SQL injection.
-        else:
-            return self._validate_fields(parsed[0])
+            raise SQLInjectionError('Invalid where clause')
 
-    def _validate_fields(self, fields):
-        query_fields = {field.replace('"', '') for field in fields if '*' not in field}
+        self._validate_fields(parsed[0])
+
+    def _validate_fields(self, fields, include_related=True):
+        if isinstance(fields, str):
+            fields = fields.split(',')
+
+        query_fields = {field.replace('"', '') for field in fields if field != '*'}
         if not query_fields:
-            return True
-        else:
-            valid_fields = set(r for r in self.related_fields).union(f['name'] for f in self.fields)
-            return query_fields.issubset(valid_fields)
+            return
+
+        valid_fields = set(f['name'] for f in self.fields)
+        if include_related:
+            valid_fields = valid_fields.union('{0}.*'.format(r.related_title) for r in self.relations)
+            valid_fields = valid_fields.union(f for f in self.related_fields)
+
+        invalid_fields = query_fields.difference(valid_fields)
+        if invalid_fields:
+            raise InvalidFieldsError('Invalid fields: {0}'.format(', '.join(invalid_fields)))
```

Comment:
Yeah, that sounds like a good way to go (though my last comment just suggested removing an error type, so it feels odd).